src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/PerformanceAnalyzerApp.java

```diff
@@ -107,21 +107,26 @@ public static void main(String[] args) {
         (MetricsConfiguration.CONFIG_MAP.get(StatsCollector.class).samplingInterval) / 2,
         TimeUnit.MILLISECONDS);
```

Comment:
Move lines 105-108 into it if (ConfigStatus...) block so that we don't start any threads if we don't have a valid config. Can you also return -1 or some error code in that case and verify that the program exits when the config is invalid?

You should be able to do this with a simple test that executes PerformanceAnalyzerApp.main(...) and checks that the return code is the error code.